keps/sig-node/2008-forensic-container-checkpointing/README.md

```diff
@@ -111,6 +118,34 @@ For the first implementation we do not want to support restore in the
 outside of Kubernetes. The restore is a container engine only operation
 in this first step.
 
+A high level view on the implementation is that triggering the *kubelet* API
+endpoint will trigger the `ContainerCheckpoint` CRI API endpoint to create a
+checkpoint at the location defined by the *kubelet*.
+
+To trigger a checkpoint following HTTP Request has to be made against the *kubelet*:
+
+- `POST /checkpoint/{namespace}/{pod}/{container}``
+- Parameters
+  - namespace (in path): string, required, Namespace
+  - pod (in path): string, required, Pod
+  - container (in path): string, required, Container
+  - timeout (in query): integer, Timeout in seconds to wait until the checkpoint
+  creation is finished. If zero or no timeout is specified the default CRI
+  timeout value will be used. Checkpoint creation time depends directly on the
+  used memory of the container. The more memory a container uses the more time
+  is required to create the corresponding checkpoint.
+- Response
+  - 200: OK
+  - 401: Unauthorized
+  - 404: Not Found (if the ContainerCheckpoint feature gate is disabled)
+  - 404: Not Found (if the specified namespace, pod or container cannot be found)
+  - 500: Internal Server Error (if the CRI implementation encounter an error during checkpointing (see error message for further details))
+  - 500: Internal Server Error (if the CRI implementation does not implement the checkpoint CRI API (see error message for further details))
+
+Permissions to this API endpoint are controlled by [Kubelet authentication/authorization](https://kubernetes.io/docs/reference/access-authn-authz/kubelet-authn-authz/)
```

Comment:
AFAIK we should mention that the kubelet api is usually restricted to cluster admins. I don't think users will have access to this.

Reply:
Actually this should be a non-goal for this feature. We are restricting this API to admins of a cluster due to security concerns around memory pages. And we hope to introduce a new KEP to allow for user checkpointing (and maybe restore) but we probably will need to move that out of kubelet so we can have proper RBAC for the checkpoint API. This is something that we can't provide as a kubelet API.

Reply:
I tried to change it based on your suggestions.